src/common/types/hugeint.cpp

```diff
@@ -291,8 +292,7 @@ bool Hugeint::TryMultiply(hugeint_t lhs, hugeint_t rhs, hugeint_t &result) {
 }
 
 // Multiply without overflow check
-template <>
-hugeint_t Hugeint::Multiply<false>(hugeint_t lhs, hugeint_t rhs) {
+hugeint_t Hugeint::SlowMultiply(hugeint_t lhs, hugeint_t rhs) {
```

Comment:
Why do we need separate `Multiply`/`SlowMultiply` implementations?

Reply:
The 2 implementations always existed, but how they are split in 2 functions. The reason this is that if we have the fast version, we want this to be inlined, and thus it must be in the header file. The fast version is very simple, and doesnt take much compile time, even when included a lot of time it should not be notcable.
The slow version is slow, so inlining doesn't help much. It is however very complex, and takes much more time to compile. If this we put this in the header file, this would increase compile times (on compilers that don't support 128 bit), since it has to be compiled every time it is included in another file.

The solution is to compile the slow code once in the .cpp, and, if available, compile the fast version on every include (but since this is very simple code it doesn't affect compile times much).